### ISSUE TITLE
insert data support xls

### ISSUE THREAD

**Issue by whwh656** [AutomaApp/automa#966]
how can i use it when read file as base64

**Comment by Kholid060** [comment]
If you want to read XLS you can use the JS block and add [sheetjs library](https://sheetjs.com/) as preload script. For example,
In the Insert data block
![image](https://user-images.githubusercontent.com/22908993/204232075-e0435e28-16e0-427b-b0f3-798b9ea40780.png)

In the JS block
![image](https://user-images.githubusercontent.com/22908993/204232113-ee12881d-d648-401a-ac42-2161bbc33839.png)

`https://cdn.sheetjs.com/xlsx-latest/package/dist/xlsx.full.min.js`

```js
let base64XLS = automaRefData('variables', 'base64XLS');
base64XLS = base64XLS.slice(base64XLS.indexOf(','));

const wb = XLSX.read(base64XLS, { type: 'base64' });
console.log(wb)
wb.SheetNames.forEach((sheetName) => {
  const values = XLSX.utils.sheet_to_json(wb.Sheets[sheetName], { header: 1 });
  console.log(`${sheetName}: `, values);
})
```

**Comment by whwh656** [comment]
http block&nbsp; support variables







 

&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "AutomaApp/automa"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年12月9日(星期五) 晚上6:29
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [AutomaApp/automa] insert data support xls (Issue #966)





 
Closed #966 as completed.
